tests/test_agent_service.py

```
@@ -100,6 +100,85 @@ async def _fake_run_async(*args, **kwargs):
+    assert "UPCOMING_EVENTS:" in prompt
+    assert "Client pitch" in prompt
+    assert "workwear" in prompt
+
+
+@pytest.mark.asyncio
+async def test_run_outfit_agent_omits_upcoming_events_when_empty():
+    from server.services.agent_service import run_outfit_agent
+
+    captured: dict = {}
+
+    async def _fake_run_async(*args, **kwargs):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_fake_run_async`

(ANN202)